core/src/main/scala/kafka/server/AbstractFetcherManager.scala

```diff
@@ -76,6 +76,17 @@ abstract class AbstractFetcherManager[T <: AbstractFetcherThread](val name: Stri
     Map("clientId" -> clientId)
   )
 
+  newGauge("DeadThreadCount", {
+    new Gauge[Int] {
+      def value: Int = {
+        deadThreadCount
+      }
+    }
+  }, Map("clientId" -> clientId))
+
+  // visible for testing
+  private[server] def deadThreadCount = lock synchronized { fetcherThreadMap.count(_._2.isThreadFailed) }
```

Comment:
I think this would be a little clearer:
```scala
fetcherThreadMap.values.count(_.isThreadFailed)
```
Also, can you annotate the return type?